### PR TITLE
[proposal] extract auth wrapper

### PR DESCRIPTION
**Pull request by Krisztiaan**
This is in no way ready, just a starting point for further discussion.

Paging @dac09 and @peterp

### CODE REVIEW

File: packages/auth/src/AuthProvider.tsx [hunk 1]
```diff
@@ -1,34 +1,30 @@
...
+  logIn: WrappedClient['logIn']
+  logOut: WrappedClient['logOut']
+  signUp: WrappedClient['signUp']
   getToken(): Promise<null | string>
   /**
    * Fetches the "currentUser" from the api side,
    * but does not update the current user state.
    **/
-  getCurrentUser(): Promise<null | CurrentUser>
+  getCurrentUser(): WrappedClient['getUserMetadata']
```
Comment (olance) [inline packages/auth/src/AuthProvider.tsx:27]: My understanding is that `currentUser` and `userMetadata` are different things?
User metadata are returned by the `AuthClient` instance, whereas the "current user" data is returned by the API

**Comment by olance** [comment]
Ah, I see why you were interested in my questions around async initialization of clients 🙃

File: packages/auth/src/AuthProvider.tsx (hunk 1 above)
Comment (Krisztiaan) [reply packages/auth/src/AuthProvider.tsx:27]: Oops, yeah, that was some overbearing typing :)

**Comment by peterp** [comment]
This is looking really good 👌

**Comment by Krisztiaan** [comment]
Alright, I'll try cleaning it up then, since this doesn't even compile, and let's see a proper implementation.

**Comment by peterp** [comment]
@Krisztiaan Cool, cool, cool - Just wanted to point out that it would be great if there was a path for people using this that didn't introduce a breaking change. Transitioning to a new implementation

**Comment by Krisztiaan** [comment]
It can be made backwards compatible and marked as `@deprecated` but I'm wondering if it's worth saving users from 2 lines of change.

**Comment by Krisztiaan** [comment]
Let's close this one for housekeeping, as there's a forum post about it now, and may be picked up later with something more in-line with any changes since the inception of this PR.

**Pull request closed without merging**